fri/src/fold_even_odd.rs

```diff
@@ -14,8 +15,9 @@ use tracing::instrument;
 /// ```ignore
 /// p_even(x) + beta p_odd(x)
 /// ```
+/// Expects input to be bit-reversed evaluations.
 #[instrument(skip_all, level = "debug")]
-pub fn fold_even_odd<F: TwoAdicField>(poly: &[F], beta: F) -> Vec<F> {
+pub fn fold_even_odd<F: TwoAdicField>(poly: Vec<F>, beta: F) -> Vec<F> {
```

Comment:
sorry for getting rid of your nice packed code, it's harder to split on even/odd now that the input is bit-reversed, and it doesn't matter much anyway because this is extension math

Reply:
Oh right, I guess it wasn't really vectorized since we don't have true packed extension fields (we can have extensions over packed base fields, but this wasn't doing that).

Might be worth looking into vectorizing it at some point, but not very important compared to getting things working